src/ios/OpenTokPlugin.m

```diff
@@ -368,20 +368,109 @@ - (void)subscriberDidConnectToStream:(OTSubscriberKit*)sub{
     NSMutableDictionary* eventData = [[NSMutableDictionary alloc] init];
     NSString* streamId = sub.stream.streamId;
     [eventData setObject:streamId forKey:@"streamId"];
-    [self triggerJSEvent: @"sessionEvents" withType: @"subscribedToStream" withData: eventData];
-
+    [self triggerJSEvent: @"subscriberEvents" withType: @"connected" withData: eventData];
+    [self triggerJSEvent: @"sessionEvents" withType: @"subscribedToStream" withData: eventData]; // Backwards compatibility
 }
-- (void)subscriber:(OTSubscriber*)subscrib didFailWithError:(OTError*)error{
+- (void)subscriberDidDisconnectFromStream:(OTSubscriberKit*)sub{
+    NSMutableDictionary* eventData = [[NSMutableDictionary alloc] init];
+    NSString* streamId = sub.stream.streamId;
+    [eventData setObject:streamId forKey:@"streamId"];
+    [self triggerJSEvent: @"subscriberEvents" withType: @"disconnected" withData: eventData];
+}
+- (void)subscriber:(OTSubscriber*)sub didFailWithError:(OTError*)error{
     NSLog(@"subscriber didFailWithError %@", error);
     NSMutableDictionary* eventData = [[NSMutableDictionary alloc] init];
-    NSString* streamId = subscrib.stream.streamId;
+    NSString* streamId = sub.stream.streamId;
     NSNumber* errorCode = [NSNumber numberWithInt:1600];
     [eventData setObject: errorCode forKey:@"errorCode"];
     [eventData setObject:streamId forKey:@"streamId"];
     [self triggerJSEvent: @"sessionEvents" withType: @"subscribedToStream" withData: eventData];
 }
+- (void)subscriberVideoDataReceived:(OTSubscriber*)sub{
+    [self triggerJSEvent: @"subscriberEvents" withType: @"videoDataReceived" withData: nil];
+}
+- (void)subscriberVideoDisabled:(OTSubscriberKit*)sub reason:(OTSubscriberVideoEventReason)reason{
+    NSMutableDictionary* eventData = [[NSMutableDictionary alloc] init];
+    NSString* reasonData = @"";
+    if(reason == OTSubscriberVideoEventPublisherPropertyChanged) {
+        reasonData = @"publishVideo";
+    } else if(reason == OTSubscriberVideoEventSubscriberPropertyChanged) {
+        reasonData = @"subscribeToVideo";
+    } else if(reason == OTSubscriberVideoEventQualityChanged) {
+        reasonData = @"quality";
+    }
+
+    [eventData setObject: reasonData forKey:@"reason"];
+    [self triggerJSEvent: @"subscriberEvents" withType: @"videoDisabled" withData: eventData];
+}
+- (void)subscriberVideoDisableWarning:(OTSubscriberKit*)sub{
+    [self triggerJSEvent: @"subscriberEvents" withType: @"videoDisableWarning" withData: nil];
+}
+- (void)subscriberVideoDisableWarningLifted:(OTSubscriberKit*)sub{
+    [self triggerJSEvent: @"subscriberEvents" withType: @"videoDisableWarningLifted" withData: nil];
+}
+- (void)subscriberVideoEnabled:(OTSubscriberKit*)sub reason:(OTSubscriberVideoEventReason)reason{
+    NSMutableDictionary* eventData = [[NSMutableDictionary alloc] init];
+    NSString* reasonData = @"";
+    if(reason == OTSubscriberVideoEventPublisherPropertyChanged) {
+        reasonData = @"publishVideo";
```

Comment:
can we create a helper function that checks the reason b/c we're duplicating code in `subscriberVideoEnabled` and `subscriberVideoDisabled`

Reply:
This has changed, dont know why its not outdated.

Reply:
@wolfenrain sorry, not sure I understand.  Can you clarify?

Reply:
Yea sorry, I have implemented the change as you requested. But Github didnt properly see the change. Therefore the review did not get put into `outdated` mode

Reply:
@wolfenrain Hmm, I just looked at your last commit and I'm not seeing any changes to the `subscriberVideoEnabled` method. https://github.com/opentok/cordova-plugin-opentok/pull/35/commits/85a19fa1eb0d03ffe56f59642d37013f2fe3c919

Can you check your local version and update as needed?